FINAL_COUNTDOWN.md

```diff
@@ -0,0 +1,75 @@
+This is the burn down issue to ship Glimmer 2. While there are still in flight PRs, this is known state of the world at SHA f1416d78c5f2fe3668e807b21d6508a164dac98c.
+
+![Awesome 80s hair band](http://vignette2.wikia.nocookie.net/degrassi/images/d/d4/Final-Countdown.gif)
+
+# Macro Tasks
+- [ ] Complete all test migration described in #13127
+- [ ] Migrate/coalesce fragment tests #13152
+
+# Features
+
+### Infra
+- [ ] Integrate into run-loop
+
+### Helpers
+- [ ] Implement `{{partial}}` [Failing Tests](https://github.com/emberjs/ember.js/blob/f1416d78c5f2fe3668e807b21d6508a164dac98c/packages/ember-glimmer/tests/integration/helpers/partial-test.js)
+- [ ] Implement `{{mut}}`
+- [ ] Implement `{{textarea}}`: [Failing test 1](https://github.com/emberjs/ember.js/blob/b418b95ea3d31e3a3656906e6ad784422f648c31/packages/ember-glimmer/tests/integration/helpers/text-area-test.js#L15) [Failing test 2](https://github.com/emberjs/ember.js/blob/b418b95ea3d31e3a3656906e6ad784422f648c31/packages/ember-glimmer/tests/integration/helpers/text-area-test.js#L23) [Failing test 3](https://github.com/emberjs/ember.js/blob/b418b95ea3d31e3a3656906e6ad784422f648c31/packages/ember-glimmer/tests/integration/helpers/text-area-test.js#L36)
+- [ ] Implement `{{each-in}}` [Failing tests](packages/ember-glimmer/tests/integration/syntax/each-in-test.js#L23)
+- [ ] Implement `{{action}}`
+
+### Currly Components
+- [ ] Implement remaining life-cycle hooks
+- [ ] Implement custom Ids for curly component invocations: [Failing test](https://github.com/emberjs/ember.js/blob/b418b95ea3d31e3a3656906e6ad784422f648c31/packages/ember-glimmer/tests/integration/components/curly-components-test.js#L26)
+- [x] Implement triple curiles #13329
+- [x] Implement `htmlSafe` #13329
```

Comment:
You should look here: https://github.com/tildeio/glimmer/blob/master/packages/glimmer-runtime/lib/compiled/opcodes/content.ts

1. The `AppendOpcode` is responsible for inserting the content during "initial render" (but this also applies to things like an `{{#if}}` changing from true to false, thus tearing down the block and rendering a different one). It needs to be updated to handle both safe and non-safe strings.

2. The `AppendOpcode` emits an `UpdateAppendOpcode` that is responsible for keeping the curlies up-to-date when the content changes. It needs to remember which "mode" it was in previously and do the right thing. (Handling safe <-> non-safe strings, etc).

3. You can look at how the "Trusting" versions of opcodes, which are for triple-curlies. You can probably refactor things to share more code.

4. Keep in mind the "const reference optimization" as you do this: https://github.com/tildeio/glimmer/pull/109

5. Future work: we also need to support appending DOM nodes into the curlies (helpers etc can return DOM nodes to be inserted). But we can tackle that separately.

6. Ideally, we would like to minimize the "mode switching" stuff by emitting specialized opcodes, but I couldn't think of any cases where we can know these information ahead of time (other than the const case, which simply does not emit updating opcodes).